Sources/Internal/SecItem.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  SecItem.swift
+//  Valet
+//
+//  Created by Dan Federman and Eric Muller on 9/16/17.
+//  Copyright © 2017 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+
+internal func execute<ReturnType>(in lock: NSLock, block: () -> ReturnType) -> ReturnType {
+    lock.lock()
+    defer {
+        lock.unlock()
+    }
+    return block()
+}
+
+
+internal final class SecItem {
+    
+    // MARK: Internal Enum
+    
+    internal enum DataResult<SuccessType> {
+        case success(SuccessType)
+        case error(OSStatus)
+    }
+    
+    internal enum Result {
```

Comment:
We originally had just `Result<SuccessType>`, but separated that out into `Result` and `DataResult<SuccessType` in 729b61811442413249fcc3062508994fad1d98b2. We were optimizing for simpler switch/case statements over fewer types. Since this API is entirely internal, having more types isn't a detriment to the overall legibility I don't think.